java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/buildsteps/docker/StageDockerArtifactBuildStep.java

```diff
@@ -1,151 +0,0 @@
-/*
- * Copyright 2017 Google Inc. All Rights Reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *       http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package com.google.cloud.runtimes.builder.buildsteps.docker;
-
-import com.google.cloud.runtimes.builder.buildsteps.base.BuildStep;
-import com.google.cloud.runtimes.builder.buildsteps.base.BuildStepException;
-import com.google.cloud.runtimes.builder.buildsteps.base.BuildStepMetadataConstants;
-import com.google.cloud.runtimes.builder.config.domain.RuntimeConfig;
-import com.google.cloud.runtimes.builder.exception.ArtifactNotFoundException;
-import com.google.cloud.runtimes.builder.exception.TooManyArtifactsException;
-import com.google.inject.Inject;
-
-import com.google.inject.assistedinject.Assisted;
-
-import org.apache.commons.io.FileUtils;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import java.io.BufferedWriter;
-import java.io.IOException;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.util.List;
-import java.util.Map;
-import java.util.stream.Collectors;
-
-public class StageDockerArtifactBuildStep extends BuildStep {
-
-  private static final String DOCKER_STAGING_DIR = ".docker_staging";
```

Comment:
We already fail if a Dockerfile is present. 

Good point regarding `.dockerignore` - a couple options come to mind:
- parse the existing dockerignore and only add new entries if they aren't already matched by what's in the existing file
- do a "dumber" parsing of the dockerignore, where we only append lines to the file that don't currently have an exact match (i.e. if the file already contains `*.yaml`, we would still append `app.yaml` even though it's redundant).

I'm leaning toward option 2.